File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int low=0;
        int high=nums.size()-1;
        int mini=INT_MAX;

        while(low<=high){
            int mid=low+(high-low)/2;

            //if left part sorted
            if(nums[low]<=nums[mid]){
                mini=min(mini,nums[low]);
                low=mid+1; // eliminate left part
            }

            else{ //right part sorted
                mini=min(mini,nums[mid]);
                high=mid-1; //eliminate right part
            }
        }
        return mini;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        low = 0
        high = len(nums) - 1
        ans = float("inf")

        while low <= high:
            mid = low + (high - low) // 2

            if nums[low] <= nums[mid]:
                ans = min(ans, nums[low])
                low = mid + 1
            else:
                ans = min(ans, nums[mid])
                high = mid - 1
        return ans
